test.js

```diff
@@ -16,6 +16,14 @@ const NO_NEWLINES_REGEXP = /^[^\n]*$/;
 const STDERR_STDOUT_REGEXP = /stderr[^]*stdout/;
 const TIMEOUT_REGEXP = /timed out after/;
 
+const waitForProcessLaunch = subprocess => {
+	return new Promise(resolve => {
+		subprocess.once('message', _ => {
```

Comment:
This could be simplified to:

```js
await promisify(subprocess.once.bind(subprocess))('message')
```

Reply:
Oh yes. I didn't think this would work. Was expecting a cb like behavior for `util.promisify` to work.
I changed it in the `waitForProcessLaunch` function and not where it's used (might looks scary for some):
79c3411a8c42db38af115ebf46c791ce78f1361b

Reply:
Yes I agree it looks a little scary. Hopefully we get the bind operator in JavaScript soon so this can become:

```js
await promisify(subprocess::once)('message')
```